art/estimators/classification/keras.py

```diff
@@ -237,7 +242,7 @@ def _initialize_params(
             "__name__" in dir(loss_function)
             and loss_function.__name__
             in ["categorical_hinge", "categorical_crossentropy", "binary_crossentropy", "kullback_leibler_divergence",]
-        ) or (self.is_tensorflow and flag_is_instance):
```

Comment:
Is removing `self.is_tensorflow` required for this pull request? I'm still concerned that it was required here.

Reply:
Yes, it is required for the tests to pass. Look at the following elif (line 250), it does a very similar check (checks if loss_function is an instance of a particular class) but does not also force self.is_tensorflow to be true. When using "regular" keras (i.e., import keras and not import tensorflow.keras) these loss classes still exist and when using them without this change you get to the raise ValueError("Loss function not recognised.").